File: benchmarks/python/split_join.py
import sys

def main():
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 200000
    parts = []
    for i in range(n):
        parts.append("item" + str(i))
    text = ",".join(parts)
    back = text.split(",")
    total = 0
    for i in range(len(back)):
        total = total + len(back[i])
    print(len(back), total, len(text))

main()

File: benchmarks/javascript/split_join.js
const n = Number(process.argv[2] ?? 200000);
const parts = [];
for (let i = 0; i < n; i++) {
    parts.push("item" + i);
}
const text = parts.join(",");
const back = text.split(",");
let total = 0;
for (let i = 0; i < back.length; i++) {
    total = total + back[i].length;
}
console.log(back.length, total, text.length);
